relay-system/src/lib.rs

```diff
@@ -16,5 +16,6 @@
 
 pub mod compat;
 mod controller;
+pub mod service;
```

Comment:
In this case, I'd rather make this a private module and do a blanket export like for `controller`. This module will contain the most used types of this crate, so it's worth having them in the root namespace.